src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,268 @@
+
+                BufferedReader reader = new BufferedReader(new FileReader(file));
+                while ((strCurrentLine = reader.readLine()) != null) {
+                    String record [] = strCurrentLine.split(" \\| ");
+                    if(record[0].equals("T")){
+                        Todo todo = new Todo(record[2]);
+                        todo.isChecked = Boolean.parseBoolean(record[1]);
+                        inventory.add(todo);
+                    }else if(record[0].equals("D")){
+                        Deadline deadline = new Deadline(record[2], record[3]);
+                        deadline.isChecked = Boolean.parseBoolean(record[1]);
+                        inventory.add(deadline);
+                    }else if(record[0].equals("E")){
+                        Event event = new Event(record[2], record[3]);
+                        event.isChecked = Boolean.parseBoolean(record[1]);
+                        inventory.add(event);
+                    }
+
+                }
+            } catch (FileNotFoundException e) {
+                System.out.println("File not found !\n");
+            }catch (IOException e) {
+                System.out.println("File cannot be opened !\n");
+            }
+        }
+        return inventory;
+    }
+    public static void main(String[] args) {
+        String intro = "Hello! I'm Duke\n" +
+                "What can I do for you?\n";
+        String bye = "Bye. Hope to see you again soon!\n";
+        ArrayList<Task> inventory = new ArrayList<Task>(100);
+        inventory = readFromFile(inventory);
+        System.out.println(intro);
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+
+        // Takes in user input until "bye" is entered by user
+        while (!line.equals("bye")){
+            try {
+                if (line.equals("list")){
+                    // List inventory
+                    listInventory(inventory);
+                }else if(line.startsWith("done")){
+                    // check items
+                    checkInventory(line, inventory);
+                } else if(line.startsWith("todo")){
+                    // Add items to Todo list
+                    todoTask(line, inventory);
+                } else if (line.startsWith("deadline")){
+                    //Add items to Deadline list
+                    deadlineTask(line, inventory);
+                } else if (line.startsWith("event")){
+                    //Add items to Event list
+                    eventTask(line, inventory);
+                }else if (line.startsWith("delete")){
+                    //Delete items in Inventory
+                    deleteItem(line, inventory);
+                }else if(line.equals("save")){
+                    //Save records to file
+                    saveToFile(inventory);
+                }
+                else{
+                    //Invalid command
+                    throw new DukeException();
+                }
```

Comment:
1. Consider using a switch statement instead
2. Excessive comments for self explanatory code can be removed